include/nuttx/tls.h

```diff
@@ -150,6 +150,7 @@ struct tls_info_s
 #endif
 
   int tl_errno;                        /* Per-thread error number */
+  uint8_t tl_data[0];
```

Comment:
change 0 to 1 since C89 doesn't support zero length array.

Reply:
I think C89 does not support `uint8_t tl_data[];` for zero length arrays and `uint8_t tl_data[0];` is a way to keep compatibility.

Reply:
unfortunately, zero length array is gnu extension, not part of any C standard:
https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html

Reply:
Guard this definition with CONFIG_SCHED_THREAD_LOCAL